### PR TITLE
added descendent venue events to sovereign venue event list

### PR DESCRIPTION
**Pull request by alexey-petrov**
Closes https://github.com/sparkletown/internal-sparkle-issues/issues/964

### CODE REVIEW

File: src/types/venues.ts [hunk 1]
```diff
@@ -11,7 +11,6 @@ import { Poster } from "./posters";
 import { Quotation } from "./Quotation";
 import { Room } from "./rooms";
 import { Table } from "./Table";
-import { UpcomingEvent } from "./UpcomingEvent";
```
Comment (sunny-viktoryia) [inline src/types/venues.ts:14]: I believe everything that is related to the `UpcomingEvent` type is legacy. Can you remove that?

File: src/types/venues.ts [hunk 2]
```diff
@@ -141,7 +140,7 @@ export interface BaseVenue {
   code_of_conduct_questions: Question[];
   owners: string[];
   iframeUrl?: string;
-  events?: Array<UpcomingEvent>; //@debt typing is this optional? I have a feeling this no longer exists @chris confirm
+  events?: WithId<VenueEvent>[]; //@debt typing is this optional? I have a feeling this no longer exists @chris confirm
```
Comment (sunny-viktoryia) [inline src/types/venues.ts:143]: events are not stored in venues anymore. So, please, remove that logic.

File: src/types/venues.ts [hunk 3]
```diff
@@ -236,6 +235,7 @@ export interface PartyMapVenue extends BaseVenue {
   admin_password?: string;
   owners: string[];
   rooms?: Room[];
+  events: WithId<VenueEvent>[];
```
Comment (sunny-viktoryia) [inline src/types/venues.ts:238]: 1.  why it was added to PartyMap when BaseVenue has already events?
2. I believe we should better handle events separately and not mutate the object by adding events there, as we don't store the information about events in the firestore-venue-object.

File: src/pages/VenueLandingPage/VenueLandingPage.tsx [hunk 4]
```diff
@@ -16,13 +18,27 @@ import VenueLandingPageContent from "./VenueLandingPageContent";
...
+  const venueEvents = useSelector(venueEventsSelector);
+  const allVenueEvents = useMemo(
+    () => [
+      ...(venueEvents || []),
+      ...(sovereignVenueId === venueId
+        ? [...descendantVenues.flatMap((venue) => venue.events || [])]
+        : []),
+    ],
+    [descendantVenues, venueEvents, sovereignVenueId, venueId]
+  );
```
Comment (sunny-viktoryia) [inline src/pages/VenueLandingPage/VenueLandingPage.tsx:40]: I believe you should use `useVenueEvents({
    venueIds: [currentVenueId, ...descedantVenueIds]
  })` here instead.

File: src/pages/Admin/EventsComponent.tsx [hunk 5]
```diff
@@ -27,30 +28,43 @@ const EventsComponent: React.FC<EventsComponentProps> = ({
...
+              ((ev?.start_utc_seconds || 1) +
+                (ev?.duration_minutes || 2) * 60) *
+                1000 >
               Date.now()
           )
-        : events,
-    [events, filterPastEvents]
+        : allVenueEvents,
+    [filterPastEvents, allVenueEvents]
   );
```
Comment (sunny-viktoryia) [inline src/pages/Admin/EventsComponent.tsx:68]: i believe you can use here something from `utils/event`. For example, will `isEventLiveOrFuture` fit here?

File: src/hooks/useSovereignVenue.ts [hunk 6]
```diff
@@ -44,9 +44,15 @@ export const useSovereignVenue: ReactHook<
...
-    if (!venueId || sovereignVenue || isSovereignVenueLoading || errorMsg)
+    if (
+      !venueId ||
+      stateVenueId === venueId ||
+      isSovereignVenueLoading ||
+      errorMsg
+    )
       return;
 
+    updateVenueId(venueId);
```
Comment (sunny-viktoryia) [inline src/hooks/useSovereignVenue.ts:55]: can you explain the logic in this file, please? why is it needed?

File: src/api/venue.ts [hunk 7]
```diff
@@ -110,12 +112,32 @@ export const fetchDirectChildVenues = async (
...
+              return {
+                ...venue,
+                ...(rest.length
+                  ? { events: rest as WithId<VenueEvent>[] }
+                  : null),
+              };
+            })
+        );
+
+        return venueWithEvents;
```
Comment (sunny-viktoryia) [inline src/api/venue.ts:140]: looks complicated... are you extending the venue with its events here? 
not sure we should do that. Can we handle that separately?

**Review by sunny-viktoryia — COMMENTED** [review]
partial review. Please, fix them and then will do another round. 
\+ i can't find the event I used to have in staging.
image
image

File: src/types/venues.ts (hunk 1 above)
Comment (alexey-petrov) [reply src/types/venues.ts:14]: It's being used in `UpcomingTickets` component. Can we confirm that it's not necessary anymore?

Comment (sunny-viktoryia) [reply src/types/venues.ts:14]: 99% - `UpcomingTickets` are legacy. You can remove them. Confirming.
(clarified with @mike-lvov )

**Pull request closed without merging**